nf_core/schema.py

```diff
@@ -613,6 +613,29 @@ def get_wf_params(self):
                 )
             )
 
+    def remove_schema_empty_definitions(self):
+        """
+        Go through top-level schema remove definitions that don't have
+        any property attributes
+        """
+        # Make copy of schema
+        schema_no_empty_definitions = copy.deepcopy(self.schema)
+
+        ## Identify and remove empty definitions from the schema
+        empty_definitions = []
+        for d_key, d_schema in list(schema_no_empty_definitions.get("definitions", {}).items()):
+            if not d_schema["properties"]:
+                del schema_no_empty_definitions["definitions"][d_key]
+                empty_definitions.append(d_key)
+
+        # Remove "allOf" group with empty definitions from the schema
+        for d_key in empty_definitions:
+            allOf = {"$ref": "#/definitions/{}".format(d_key)}
```

Comment:
Working on this PR at the moment so no need to change, but FYI we are trying to slowly move everything to the new format string style in the codebase now (with a few exceptions where this old style is still clearer). Generally it's much easier to read and maintain.

So here it'd be:
```suggestion
            allOf = {"$ref": f"#/definitions/{d_key}"}
```

Just prepend the string quotes with `f` and then stick the variable name into the squiggly brackets.